gazebo_plugins/src/gazebo_ros_ackermann_drive.cpp

```diff
@@ -355,7 +358,7 @@ void GazeboRosAckermannDrive::Load(gazebo::physics::ModelPtr _model, sdf::Elemen
   impl_->last_update_time_ = _model->GetWorld()->SimTime();
 
   impl_->cmd_vel_sub_ = impl_->ros_node_->create_subscription<geometry_msgs::msg::Twist>(
-    "cmd_vel", rclcpp::QoS(rclcpp::KeepLast(1)),
+    "cmd_vel", qos.get_subscription_qos("cmd_vel", rclcpp::QoS(1)),
```

Comment:
It would be nice to avoid the repetition of the topic name here.
I'm not sure what is the right way of doing that, is overriding `create_publisher` and `create_subscription` in `gazebo_ros::Node` acceptable?

If not, `get_subscription_qos` could return a `struct` composed by the `rclcpp::QoS` object and the topic name.
The final code would look like:

```cpp
  auto result = qos.get_subscription_qos("cmd_vel", rclcpp::QoS(1));
  impl_->cmd_vel_sub_ = impl_->ros_node_->create_subscription<geometry_msgs::msg::Twist>(
    result.topic_name, result.qos,
    std::bind(&GazeboRosAckermannDrivePrivate::OnCmdVel, impl_.get(), std::placeholders::_1));
```

I don't have a strong opinion about this, the repetition isn't too bad either.

Reply:
I thought about overriding `create_publisher` and `create_subscription`, but I thought it seemed a little invasive and could take away from the readability of the code. I think it's clear what's happening when we get the QoS from the `gazebo_ros::QoS` object compared to hiding it in a call to `create_publisher`.

On the other hand, maybe it's fine to hide this detail from the plugin developer. Then all plugins will have configurable QoS without needed to opt-in.

Reply:
> I thought about overriding create_publisher and create_subscription, but I thought it seemed a little invasive and could take away from the readability of the code. I think it's clear what's happening when we get the QoS from the gazebo_ros::QoS object compared to hiding it in a call to create_publisher.

Yes, I have a similar feeling about overriding.
I won't block on this, you can decide what you think is best.

---

Besides my strange idea of modifying `get_subscription_qos` return value, another option is:

```cpp
  std::string topic_name = "cmd_vel";
  impl_->cmd_vel_sub_ = impl_->ros_node_->create_subscription<geometry_msgs::msg::Twist>(
    topic_name, qos.get_subscription_qos(topic_name, rclcpp::QoS(1)),
    std::bind(&GazeboRosAckermannDrivePrivate::OnCmdVel, impl_.get(), std::placeholders::_1));
```

There's no great benefit either.

Reply:
I'd like to try overriding `create_publisher` and `create_subscription` to see what it looks like; whenever I get a moment. Feel free to propose the change yourself if you have time (either updating this branch or targeting a PR against it).

Reply:
We also have things like `image_transport::create_publisher`, and I'm not sure if overriding `Node` methods will be enough (maybe, we're passing the NodeBase interface ....).

I will prefer to merge this as-is, and we can do in improvement later.
What do you think?

Reply:
SGTM